### PR TITLE
Use boost::thread_specific_ptr instead of thread_local

### PR DESCRIPTION
**Pull request by seelabs**

### CODE REVIEW

**Comment by vinniefalco** [comment]
`thread_specific_ptr`

File: src/ripple/core/impl/ThreadEntry.cpp [hunk 1]
```diff
@@ -22,47 +22,49 @@
...
+    threadName.reset(new std::string(std::move(name)));
 }
 }
 
 void terminateHandler()
 {
     if (std::current_exception())
     {
+        if (!detail::threadName.get())
+            detail::threadName.reset(new std::string("Unknown"));
```
Comment (willwray) [inline src/ripple/core/impl/ThreadEntry.cpp]: detail::setThreadName("Unknown");

Comment (seelabs) [inline src/ripple/core/impl/ThreadEntry.cpp]: @willwray Done. I also made `threadName` static while I was at it (`threadName` used to be `extern thread_local std::string`, but there's a gcc bug that we use `setThreadName` to work around; so it should be static now).

**Comment by HowardHinnant** [comment]
👍

**Comment by wilsonianb** [comment]
glibc 2.18 is now longer required
A rippled rpm with this change is able to be installed on CentOS 7 with glibc 2.17
:+1:

**Pull request closed without merging**